File: backend/app/schemas/audit.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from typing import Optional, Dict, Any

class AuditLogBase(BaseModel):
    action: str = Field(..., description="The action performed, e.g., 'LOGIN', 'CREATE_RULE', 'VERIFY_DOCUMENT'")
    resource_type: Optional[str] = Field(None, description="The type of resource affected, e.g., 'RULE', 'DOCUMENT', 'SYSTEM'")
    resource_id: Optional[str] = Field(None, description="The ID of the resource affected")
    details: Optional[Dict[str, Any]] = Field(default_factory=dict, description="Additional structured details about the action")

class AuditLogCreate(AuditLogBase):
    user_id: Optional[str] = None
    ip_address: Optional[str] = None

class AuditLogResponse(AuditLogBase):
    id: str
    user_id: Optional[str]
    ip_address: Optional[str]
    created_at: datetime
    
    # Optional field to include user email when returning to frontend
    user_email: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/audit.ts
import apiClient from './client'

export interface AuditLog {
  id: string
  user_id: string | null
  user_email: string | null
  action: string
  resource_type: string | null
  resource_id: string | null
  details: Record<string, any>
  ip_address: string | null
  created_at: string
}

export interface GetAuditLogsParams {
  skip?: number
  limit?: number
  action?: string
  resource_type?: string
  days?: number
}

export const getAuditLogs = (params: GetAuditLogsParams) => {
  return apiClient.get<any, AuditLog[]>('/audit-logs/', { params })
}
